packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/services/DocumentService.java

```diff
@@ -0,0 +1,172 @@
+package software.uncharted.terarium.hmiserver.services;
+
+import software.uncharted.terarium.hmiserver.models.XDD.Document;
+import software.uncharted.terarium.hmiserver.models.XDD.Extraction;
+import software.uncharted.terarium.hmiserver.models.XDD.XDDArticlesResponseOK;
+import software.uncharted.terarium.hmiserver.models.XDD.XDDExtractionsResponseOK;
+import software.uncharted.terarium.hmiserver.models.XDD.XDDResponse;
+import software.uncharted.terarium.hmiserver.models.XDD.XDDSearchPayload;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.util.ArrayList;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@ApplicationScoped
+public class DocumentService {
+	// XDD API URLs
+	private String DOCUMENTS_BASE_URL = "https://xdd.wisc.edu/api/articles?";
```

Comment:
these hardcoded strings can by set in the application.properties file

Reply:
done